### ISSUE TITLE
Wrong http version for http/2

### ISSUE THREAD

**Issue by danielkec** [oracle/helidon#4295]
## Environment Details
* Helidon Version: 2.5.0
* Helidon SE
* JDK version: 18.ea.5.lm-open
* OS: Ubuntu 18.04.6 LTS

----------

## Problem Description
`req.version()` returns http/1.1 even on http/2 connection

## Steps to reproduce
```java
        WebServer.builder(Routing.builder()
                        .get("/test", (req, res) -> res.send("Hi! HTTP version is "+req.version()))
                        .build())
                .experimental(ExperimentalConfiguration.builder()
                        .http2(Http2Configuration.builder()
                                .enable(true)
                                .build())
                        .build())
                .port(8080)
                .host("localhost")
                .build()
                .start()
                .await();
```

```shell
$ curl -vv --http2-prior-knowledge http://localhost:8080/test
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x55998724f620)
> GET /test HTTP/2
> Host: localhost:8080
> User-Agent: curl/7.58.0
> Accept: */*
> 
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 200 
< content-type: text/plain
< date: Tue, 31 May 2022 16:16:34 +0200
< x-http2-stream-weight: 16
< content-length: 24
< 
* Connection #0 to host localhost left intact
Hi! HTTP version is V1_1
```

**Comment by danielkec** [comment]
Fixed for master in #4236

**Comment by danielkec** [comment]
Fixed for 2x with #4629
